libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/fakeResolver.kt

```diff
@@ -6,8 +6,9 @@ import com.apollographql.apollo3.api.json.MapJsonReader
  * @property path the path to the element being resolved. [path] is a list of [String] or [Int]
  * @property mergedField the field being resolved
  */
-class FakeResolverContext(
+class FakeResolverContext internal constructor(
     val path: List<Any>,
+    val cacheKey: String,
```

Comment:
I'm not a huge fan of adding the concept of cacheKey there but didn't find an alternative. 

GraphQL Java [exposes the parent `map` and path of types](https://www.graphql-java.com/documentation/data-fetching/#the-interesting-parts-of-the-datafetchingenvironment). That's enough to compute the current field parent cache key but it means that:

1. this has to be done for each leaf field
2. this has to be done for each parent up to the root

This is incredibly wasteful. The other option would be to make `FakeResolverContext` into a visitor pattern and it could then maintain its own stack in parallel to the recursion that's happening in `buildFakeObject` but that feels a bit too much details for an implementation.